Pattern Template Update via discussion w/ Bob Hanmer

nyeates opened the pull request:
Bob and I had a small side meeting to exchange understandings. I took notes of some explanations he gave for pattern fields!

Commit:
Updated the template from notes with Bob

I took notes from Bob Hanmer, clarified my understanding, and changed some other pieces that needed updating.

NewMexicoKid commented on meta/pattern-template.md:
```diff
@@ -1,32 +1,38 @@
 
 ## Context
-Where does the problem exist? what are the pre-conditions?
+Where does the problem exist? what are the pre-conditions? **Unchangable** before the solution goes into place. The content here is often tied to applicability of other readers: "Do I have this same particular situation?"
```
Unchangeable (spelling)

NewMexicoKid commented on meta/pattern-template.md:
```diff
@@ -1,32 +1,38 @@
-Verified resolutions and possible resolutions to the problem
+Verified resolutions and possible resolutions to the problem.
 
 ## Desired Resulting Context
```
Should this just be
## Resulting Context

It's only Desired Resulting Context in the case of a donut, no?

NewMexicoKid commented on meta/pattern-template.md:
```diff
@@ -1,32 +1,38 @@
 ## Problem
-What is the problem - crisp definition of the problem
+What is the problem - crisp definition of the problem. 1-2 sentances. Anymore should go into other sections.
 
```
## Story (Optional)
Sometimes there is a short story that succinctly illustrates the problem, giving useful background information.

NewMexicoKid committed:
Made a few corrections to pattern-template.md

1. Desired Resulting Context -> Resulting Context
2. Status is only optional until it is merged (at which point the PR labeling is no longer easily seen)

NewMexicoKid committed:
Fixed spelling error

Also put in optional Story field
